node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/ServiceMonitorStub.java

```diff
@@ -89,4 +90,8 @@ private ServiceStatus getHostStatus(String hostname) {
         return status;
     }
 
+    @Override
+    public ServiceModel getServiceModelSnapshot() {
+        throw new IllegalStateException("getServicemodelSnapshot has not been implemented");
```

Comment:
Consider using `UnsupportedOperationException`.

Reply:
Thanks, for some reason I unable to find a more appropriately named class. Fixed.